### PR TITLE
[Roslyn] Implement IDocumentSupportsFeatureService

### PR DESCRIPTION
**Pull request by iainx**

### CODE REVIEW

**Comment by DavidKarlas** [comment]
@iainx can you just confirm `== null` is intentional?

**Comment by Therzok** [comment]
@DavidKarlas yes, it is. If the document is registered, it's a contained language (i.e. razor)

@slluis none yet opened for it, as it isn't used yet.

The idea is that VSWindows disables the features specified in the service for Razor documents, as that's not supported in the commanding layer that will be integrated from editor features in the future.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices/MonoDevelopDocumentSupportsFeatureService.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using MonoDevelop.Ide.TypeSystem;
+
+namespace MonoDevelop.Ide.RoslynServices
+{
+	[ExportWorkspaceService (typeof(IDocumentSupportsFeatureService), ServiceLayer.Host), Shared]
+	internal sealed class MonoDevelopDocumentSupportsFeatureService : IDocumentSupportsFeatureService
+	{
+		public bool SupportsCodeFixes (Document document)
+		{
+			return (MonoDevelopHostDocumentRegistration.FromDocument (document) == null);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices/MonoDevelopDocumentSupportsFeatureService.cs]: To ensure clarity, maybe we should add a helper method like:
```
static bool IsContainedDocument (Document document)
{
    return MonoDevelopHostDocumentRegistration.FromDocument (document) == null;
}
```

and the service becomes:
```
...
public bool SupportsNavigationToAnyPosition (Document document) => !IsContainedDocument (document);
...
```

**Comment by iainx** [comment]
@Therzok updated

**Pull request merged**